core/src/main/java/io/ddf/Factor.java

```diff
@@ -99,14 +113,15 @@ public Factor(String name, T[] data) throws DDFException {
    * @param engineName
    * @throws DDFException
    */
-  public Factor(String name, T[] data, String engineName) throws DDFException {
+  public Factor(String name, T[] data, String engineName, Schema.ColumnType type) throws DDFException {
     super(name, data, engineName);
+    this.mType= type;
   }
 
 
   private Map<String, Integer> mLevelMap;
```

Comment:
Why `mLevels` is a `List<Object>` but `mLevelMap` is a map with `String` as key?